include/oneapi/dpl/pstl/glue_algorithm_ranges_impl.h

```diff
@@ -602,6 +784,264 @@ struct __merge_fn
 
 inline constexpr __internal::__merge_fn merge;
 
+// [alg.fill]
+
+namespace __internal
+{
+
+struct __fill_fn
+{
+    template<typename _ExecutionPolicy, std::ranges::random_access_range _R, typename _T = std::ranges::range_value_t<_R>>
+    requires oneapi::dpl::is_execution_policy_v<std::remove_cvref_t<_ExecutionPolicy>> &&
+             std::ranges::sized_range<_R> && std::indirectly_writable<std::ranges::iterator_t<_R>, const _T&>
+
+    std::ranges::borrowed_iterator_t<_R>
+    operator()(_ExecutionPolicy&& __exec, _R&& __r, const _T& __value) const
+    {
+        const auto __dispatch_tag = oneapi::dpl::__ranges::__select_backend(__exec);
+        return oneapi::dpl::__internal::__ranges::__pattern_fill(__dispatch_tag, std::forward<_ExecutionPolicy>(__exec),
+            std::forward<_R>(__r), __value);
+    }
+}; //__fill_fn
+
+} //__internal
+
+inline constexpr __internal::__fill_fn fill;
+
+// [alg.generate]
+
+namespace __internal
+{
+
+struct __generate_fn
+{
+    template<typename _ExecutionPolicy, std::ranges::random_access_range _R, std::copy_constructible _F>
+    requires oneapi::dpl::is_execution_policy_v<std::remove_cvref_t<_ExecutionPolicy>> && std::invocable<_F&> &&
+        std::ranges::sized_range<_R> && std::indirectly_writable<std::ranges::iterator_t<_R>, std::invoke_result_t<_F&>>
+
+    std::ranges::borrowed_iterator_t<_R>
+    operator()(_ExecutionPolicy&& __exec, _R&& __r, _F __gen) const
+    {
+        const auto __dispatch_tag = oneapi::dpl::__ranges::__select_backend(__exec);
+        return oneapi::dpl::__internal::__ranges::__pattern_generate(__dispatch_tag, std::forward<_ExecutionPolicy>(__exec),
+            std::forward<_R>(__r), __gen);
+    }
+}; //__generate_fn
+
+} //__internal
+
+inline constexpr __internal::__generate_fn generate;
+
+// [alg.move]
+
+namespace __internal
+{
+
+struct __move_fn
+{
+    template<typename _ExecutionPolicy, std::ranges::random_access_range _InRange, std::ranges::random_access_range _OutRange>
+    requires oneapi::dpl::is_execution_policy_v<std::remove_cvref_t<_ExecutionPolicy>>
+        && std::ranges::sized_range<_InRange> && std::ranges::sized_range<_OutRange>
+        && std::indirectly_movable<std::ranges::iterator_t<_InRange>, std::ranges::iterator_t<_OutRange>>
+
+    std::ranges::move_result<std::ranges::borrowed_iterator_t<_InRange>, std::ranges::borrowed_iterator_t<_OutRange>>
+    operator()(_ExecutionPolicy&& __exec, _InRange&& __r, _OutRange&& __out_r) const
+    {
+        auto [__res_in, __res_out] =
+            oneapi::dpl::ranges::transform(std::forward<_ExecutionPolicy>(__exec), std::forward<_InRange>(__r),
```

Comment:
Same comment here about calling discouraging calling public APIs.  Should we call `__pattern_transform`

Reply:
` calling discouraging calling public APIs` - could you please explain this wording?

Reply:
Regarding to a pattern call. As I commented above...
We do such way and in another algorithms and in the iterator base API as well.
It makes the code a little bit simpler and shorter.
What's wrong with such approach?

Reply:
> ` calling discouraging calling public APIs` - could you please explain this wording?

This is just a mistake.  "Same comment here about not calling public APIs".

Reply:
I suppose I can't really come up with a big downside when it is being called at this level of the code.  However we do have to be careful when doing it not to add extra overhead or loops etc. I don't see any actual problems with the cases here. Mostly, I had a memory that we should avoid this as a general rule, but perhaps I am mistaken.
I'll retract my comments about this.